File: TenThingsDemos/TenThingsDemos/10_Expressions.fs
module Expressions

open System
open System.IO

let ifExpressionTest value =
  let abs = if value < 0 then -value else value
  printfn "%d" abs

let matchExpressionTest value =
  let trollText =
    match value with
    | 0 -> "none"
    | 1 -> "one"
    | 2 -> "some"
    | 3 -> "many"
    | _ -> "lots"
  printfn "%s" trollText

let exceptionExpressionTest() =
  let text =
    try
      File.ReadAllText(@"c:\temp\test.txt")
    with
      | :? FileNotFoundException -> String.Empty
  printfn "File contents: %s" text

let usingExpressionTest() =
  let lineCount =
    using (File.OpenRead(@"c:\temp\test.txt")) <| fun stream ->
      using (new StreamReader(stream)) <| fun reader ->
        reader.ReadToEnd().Split('\n').Length
  printfn "%d" lineCount

File: TenThingsDemos/CSharpEquivalents/10_Expressions.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpEquivalents
{
    public static class Expressions
    {
        public static void IfStatementTest(int value)
        {
            int abs;

            if (value < 0)
            {
                abs = -value;
            }
            else
            {
                abs = value;
            }

            Console.WriteLine(abs);
        }

        public static void SwitchStatementTest(int value)
        {
            string trollText;

            switch (value)
            {
                case 0: trollText = "none"; break;
                case 1: trollText = "one"; break;
                case 2: trollText = "some"; break;
                case 3: trollText = "many"; break;
                default: trollText = "lots"; break;
            }

            Console.WriteLine(trollText);
        }

        public static void IfExpressionTest(int value)
        {
            int abs = value < 0 ? -value : value;
            Console.WriteLine(abs);
        }

        public static void ExceptionStatementTest()
        {
            string text;

            try
            {
                text = File.ReadAllText(@"c:\temp\test.txt");
            }
            catch (FileNotFoundException)
            {
                text = String.Empty;
            }

            Console.WriteLine("File contents: " + text);
        }

        public static void UsingStatementTest()
        {
            int lineCount;

            using (var stream = File.OpenRead(@"c:\temp\test.txt"))
            {
                using (var reader = new StreamReader(stream))
                {
                    lineCount = reader.ReadToEnd().Split('\n').Length;
                }
            }

            Console.WriteLine(lineCount);
        }

    }
}
